pkg/images/integration/set_impl.go

```diff
@@ -56,18 +56,33 @@ func (e *setImpl) UpdateImageIntegration(integration *storage.ImageIntegration)
 		return
 	}
 
+	var isRegistry bool
+	var isScanner bool
 	for _, category := range integration.GetCategories() {
 		switch category {
 		case storage.ImageIntegrationCategory_REGISTRY:
+			isRegistry = true
 			err = e.registrySet.UpdateImageIntegration(integration)
 		case storage.ImageIntegrationCategory_SCANNER:
+			isScanner = true
 			err = e.scannerSet.UpdateImageIntegration(integration)
 		case storage.ImageIntegrationCategory_NODE_SCANNER: // This is because node scanners are implemented into image integrations
 		default:
 			err = fmt.Errorf("source category %q has not been implemented", category)
 		}
 	}
 
+	// An integration may have a category removed, for example, if an integration went from being
+	// both a registry + scanner to just a registry. On update we need to remove the integration
+	// from the sets it should no longer be a part of.
+	if !isRegistry {
+		_ = e.registrySet.RemoveImageIntegration(integration.GetId())
```

Comment:
Nitpick: Can we change the interface not to return an error instead? I see this as a tech debt -- implementation might assume the error contract is respected everywhere in future changes; while clients are playing smart and ignoring it because they know it's meaningless.

Reply:
It's a bit out of your scope, hence the "nitpick" tag. Meaning, feel free to move on.

Reply:
I made this change locally to see the impact, and the current 'usages' check for the error and respond accordingly.  I'm thinking now instead of ignoring the error I should log it instead, even though for this use case it will always be `nil` - if we decide to move the backing store for these sets to be something that could error out, it would reduce the amount of change needed.

Reply:
https://github.com/stackrox/stackrox/pull/8939/commits/35f0ce0f139373869c4ef43f5205f019944a8b99

Reply:
> if we decide to move the backing store for these sets to be something that could error out, it would reduce the amount of change needed.

It's OK for now. But to clarify, moving to something that can error out could require additional changes here, probably surfacing the error. Otherwise, the caller would see this as successful, but the integration could still potentially be in the `registrySet` or `scannerSet`, depending on which one failed to remove/update. I would have to check the above layers to assess the impact.

Reply:
I think logging this is ok because the original implementation never did the deletion, so it's not a regression